tests/db_functions/test_nullif.py

```diff
@@ -0,0 +1,46 @@
+from unittest import skipIf, skipUnless
+
+from django.db import connection
+from django.db.models import Value
+from django.db.models.functions import NullIf
+from django.test import TestCase
+
+from .models import Author
+
+
+class NullIfTests(TestCase):
+
+    @classmethod
+    def setUpTestData(cls):
+        Author.objects.create(name='John Smith', alias='smithj', age=40)
+        Author.objects.create(name='Rhonda', alias='Rhonda', age=45)
+
+    @skipIf(connection.vendor == 'oracle', 'oracle uses the empty string instead')
+    def test_nullif(self):
+        authors = Author.objects.annotate(nullif=NullIf('alias', 'name')).values_list('nullif')
+        self.assertSequenceEqual(authors, [('smithj',), (None,)])
+
+    @skipUnless(connection.vendor == 'oracle', 'oracle specific test for the NULL return value')
+    def test_nullif_oracle(self):
+        authors = Author.objects.annotate(nullif=NullIf('alias', 'name')).values_list('nullif')
+        self.assertSequenceEqual(authors, [('smithj',), ('',)])
+
+    @skipUnless(connection.vendor == 'oracle', 'Test another type because empty string is treated as NULL')
+    def test_nullif_oracle_int(self):
+        authors = Author.objects.annotate(nullif=NullIf('age', Value(40))).values_list('nullif')
+        self.assertSequenceEqual(authors, [('',), (45,)])
```

Comment:
You don't need to add a Oracle-specific test for types other than `BinaryField` and `CharField`, it will work exactly the same as for other back-ends. Please see my [comment](https://github.com/django/django/pull/9543#discussion_r160059507).

Reply:
For this reason, I added a new field to the `Author` model.